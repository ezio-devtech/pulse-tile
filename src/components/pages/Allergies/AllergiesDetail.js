import React, { PureComponent } from 'react';
import { Row, Col } from 'react-bootstrap';

import AlergiesDetailPanel from './AlergiesDetailPanel'

const ALLERGIE_PANEL = 'allergiePanel';
const META_PANEL = 'metaPanel';


export default class AllergiesDetail extends PureComponent {
  render() {
    const { onExpand, name, onShow, openedPanel } = this.props
    return (
      <div className="section-detail">
        <div className="panel-group accordion">
          <AlergiesDetailPanel
            onExpand={onExpand}
            name={name}
            title="Allergy"
            onShow={onShow}
            isOpen={openedPanel === ALLERGIE_PANEL}
          >
            <div className="panel-body-inner">
              <div className="form">
                <div className="form-group-wrapper">
                  <Row>
                    <Col xs={12} md={6}>
                      <Row>
                        <div className="col-md-11">
                          <div className="form-group">
                            <label className="control-label">Cause</label>
                            <div className="form-control-static">test</div>
                          </div>

                          <div className="form-group">
                            <label className="control-label">Reaction</label>
                            <div className="form-control-static">test</div>
                          </div>

                          <div className="form-group">
                            <label className="control-label">Author</label>
                            <div className="form-control-static">test</div>
                          </div>

                          <div className="form-group">
                            <label className="control-label">Date</label>
                            <div className="form-control-static">test</div>
                          </div>

                          <div className="form-group">
                            <label className="control-label">Source</label>
                            <div className="form-control-static">test</div>
                          </div>
                        </div>
                      </Row>
                    </Col>
                  </Row>
                </div>
              </div>
            </div>
          </AlergiesDetailPanel>
          <AlergiesDetailPanel
            onExpand={onExpand}
            name={name}
            title="Edit Metadata"
            isOpen={openedPanel === META_PANEL}
            onShow={onShow}
          >
            <div className="panel-body-inner">
              <div className="form">
                <div className="form-group-wrapper">
                  <Row>
                    <Col xs={12} md={6}>
                      <Row>
                        <div className="col-md-11">
                          <div className="form-group">
                            <label className="control-label">Cause Code</label>
                            <div className="form-control-static">test</div>
                          </div>

                          <div className="form-group">
                            <label className="control-label">Terminology</label>
                            <div className="form-control-static">test</div>
                          </div>
                        </div>
                      </Row>
                    </Col>
                  </Row>
                </div>
              </div>
            </div>
          </AlergiesDetailPanel>
        </div>
      </div>
    )
  }
}
