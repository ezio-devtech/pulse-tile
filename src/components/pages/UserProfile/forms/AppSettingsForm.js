import React, { PureComponent } from 'react';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import FileInput from '../../../form-fields/FileInput';
import { optionsForThemesField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';

@reduxForm({ form: 'appSettingsFormSelector' })
export default class AppSettingsForm extends PureComponent {
  render() {
    return (
      <div className="panel-body-inner">
        <form name="appSettingsForm" className="form">
          <div className="form-group-wrapper">
            <Row>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11}>
                    <Field
                      label="Application Title"
                      name="applicationTitle"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label="Application Logo File"
                      name="logoPath"
                      component={FileInput}
                      id="logoPath"
                    />
                    <Field
                      label="Application Themes"
                      name="green"
                      component={Select}
                      options={optionsForThemesField}
                    />
                    <Field
                      label="Browser Window title"
                      name="browserWindowTitle"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
            </Row>
          </div>
        </form>
      </div>)
  }
}
