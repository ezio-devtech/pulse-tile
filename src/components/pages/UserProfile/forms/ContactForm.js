import React, { PureComponent } from 'react';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import { optionsForCountryField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';
import { validateContactForm } from './validation';

@reduxForm({
  form: 'contactFormSelector',
  validate: validateContactForm,
})
export default class ContactForm extends PureComponent {
  render() {
    return (
      <div className="panel-body-inner">
        <form name="personalForm" className="form">
          <div className="form-group-wrapper">
            <Row>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11}>
                    <Field
                      label="Address"
                      name="address"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label="City"
                      name="city"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label="State"
                      name="state"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label="Postal Code"
                      name="postalCode"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label="Country"
                      name="country"
                      placeholder=""
                      component={Select}
                      options={optionsForCountryField}
                    />
                  </Col>
                </Row>
              </Col>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11} mdOffset={1}>
                    <Field
                      label="Phone Number"
                      name="phone"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label="Email"
                      name="email"
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
            </Row>
          </div>
        </form>
      </div>)
  }
}
