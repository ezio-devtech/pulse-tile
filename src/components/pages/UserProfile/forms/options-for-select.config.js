const optionsForThemesField = [{
  value: 'green',
  title: 'Green Theme',
},
{
  value: 'purple',
  title: 'Purple Theme',
},
{
  value: 'gray',
  title: 'Gray Theme',
}];

const optionsForGenderField = [{
  value: 'female',
  title: 'Female',
}, {
  value: 'male',
  title: 'Male',
}];

export { optionsForThemesField, optionsForGenderField }
