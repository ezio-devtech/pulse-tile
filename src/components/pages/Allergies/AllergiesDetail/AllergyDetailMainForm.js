import React, { PureComponent } from 'react';
import { Field, reduxForm } from 'redux-form'

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import ValidatedTextareaFormGroup from '../../../form-fields/ValidatedTextareaFormGroup';
import DateInput from '../../../form-fields/DateInput';
import { validateAllergiesPanel } from '../forms.validation';
import { valuesNames, valuesLabels } from '../forms.config';

@reduxForm({
  form: 'allergiePanelFormSelector',
  validate: validateAllergiesPanel,
})
export default class AllergyDetailMainForm extends PureComponent {
  componentDidMount() {
    const { detail, initialize } = this.props;
    initialize(this.defaultValuesForm(detail));
  }
  defaultValuesForm(value) {
    const defaultFormValues = {
      [valuesNames.CAUSE]: value.cause,
      [valuesNames.REACTION]: value.reaction,
      [valuesNames.AUTHOR]: value.author,
    };

    return defaultFormValues;
  }
  render() {
    const { detail, isSubmit } = this.props;
    return (
      <div className="panel-body-inner">
        <form name="allergiePanelForm" className="form">
          <div className="form-group-wrapper">
            <div className="row-expand">
              <div className="col-expand-left">
                <Field
                  label={valuesLabels.CAUSE}
                  name={valuesNames.CAUSE}
                  id={valuesNames.CAUSE}
                  type="text"
                  placeholder=""
                  component={ValidatedInput}
                  props={isSubmit}
                />
                <Field
                  label={valuesLabels.REACTION}
                  name={valuesNames.REACTION}
                  id={valuesNames.REACTION}
                  component={ValidatedTextareaFormGroup}
                  props={isSubmit}
                />
              </div>
            </div>
            <div className="row-expand">
              <div className="col-expand-left">
                <Field
                  label={valuesLabels.AUTHOR}
                  name={valuesNames.AUTHOR}
                  id={valuesNames.AUTHOR}
                  component={ValidatedInput}
                  props={{ disabled: true }}
                />
              </div>
              <div className="col-expand-right">
                <Field
                  label={valuesLabels.DATE}
                  name={valuesNames.DATE}
                  id={valuesNames.DATE}
                  component={DateInput}
                  props={{ disabled: true, value: detail.dateCreated, format: 'DD-MMM-YYYY' }}
                />
              </div>
            </div>
          </div>
        </form>
      </div>)
  }
}
