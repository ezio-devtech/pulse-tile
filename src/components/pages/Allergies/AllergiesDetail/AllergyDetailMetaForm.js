import React, { PureComponent } from 'react';
import { Field, reduxForm } from 'redux-form'

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import StaticFormField from '../../../form-fields/StaticFormField';
import { validateAllergiesMeta } from '../forms.validation';
import { valuesNames, valuesLabels } from '../forms.config';

@reduxForm({
  form: 'metaPanelFormSelector',
  validate: validateAllergiesMeta,
})
export default class AllergieMetaForm extends PureComponent {
  componentDidMount() {
    const { detail, initialize } = this.props;
    initialize(this.defaultValuesForm(detail));
  }
  defaultValuesForm(value) {
    const defaultFormValues = {
      [valuesNames.CAUSECODE]: value.causeCode,
      [valuesNames.TERMINOLOGY]: value.causeTerminology,
      [valuesNames.TERMINOLOGYCODE]: value.terminologyCode,
    };

    return defaultFormValues;
  }
  render() {
    const { isSubmit } = this.props;
    return (
      <div className="panel-body-inner">
        <form name="allergieMetaForm" className="form">
          <div className="form-group-wrapper">
            <div className="row-expand">
              <div className="col-expand-left">
                <Field
                  name={valuesNames.CAUSECODE}
                  label={valuesLabels.CAUSECODE}
                  component={StaticFormField}
                  props={{ className: 'form-control-static' }}
                />
              </div>
            </div>
            <div className="row-expand">
              <div className="col-expand-left">
                <Field
                  label={valuesLabels.TERMINOLOGY}
                  name={valuesNames.TERMINOLOGY}
                  id={valuesNames.TERMINOLOGY}
                  component={ValidatedInput}
                  props={{isSubmit}}
                />
              </div>
              <div className="col-expand-right">
                <Field
                  name={valuesNames.TERMINOLOGYCODE}
                  label={valuesLabels.TERMINOLOGYCODE}
                  component={StaticFormField}
                  props={{ className: 'form-control-static' }}
                />
              </div>
            </div>
          </div>
        </form>
      </div>)
  }
}
