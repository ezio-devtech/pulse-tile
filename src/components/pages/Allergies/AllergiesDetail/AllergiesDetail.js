import React, { PureComponent } from 'react';
import { Row, Col } from 'react-bootstrap';
import { get } from 'lodash';

import PluginDetailPanel from '../../../plugin-page-component/PluginDetailPanel';
import AllergyDetailMainForm from './AllergyDetailMainForm';
import Switch from '../../../form-fields/Switch';
import { getDDMMMYYYY } from '../../../../utils/time-helpers.utils';
import { valuesNames, valuesLabels } from '../forms.config';
import { themeConfigs } from '../../../../themes.config';
import { isButtonVisible, isPanelVisible, isShowElement } from '../../../../utils/themeSettings-helper';

const ALLERGIE_PANEL = 'allergiePanel';
const META_PANEL = 'metaPanel';
const SYSTEM_INFO_PANEL = 'systemInformationPanel';

export default class AllergiesDetail extends PureComponent {
  render() {
    const { onExpand, name, onShow, openedPanel, expandedPanel, currentPanel, onEdit, editedPanel, onCancel, onSaveSettings, allergiePanelFormValues, metaPanelFormValues, isSubmit } = this.props;
		let { detail } = this.props;
		detail = detail || {};
    const dateCreated = getDDMMMYYYY(detail[valuesNames.DATE_CREATED]);

    const hideElements = get(themeConfigs, 'detailsToHide.allergies', []);
    const hiddenButtons = get(themeConfigs, 'buttonsToHide.allergies', []);
    const hiddenPanels = get(themeConfigs, 'panelsToHide.allergies', []);

    const detailsTitle = get(themeConfigs.patientsSummaryDetailsTitles, 'allergies', 'Allergy');

    return (
      <div className="section-detail">
        <div className="panel-group accordion">
          {(expandedPanel === ALLERGIE_PANEL || expandedPanel === 'all') && !editedPanel[ALLERGIE_PANEL] ? <PluginDetailPanel
            onExpand={onExpand}
            name={ALLERGIE_PANEL}
            title={detailsTitle}
            onShow={onShow}
            isOpen={openedPanel === ALLERGIE_PANEL}
            currentPanel={currentPanel}
            onEdit={onEdit}
            editedPanel={editedPanel}
            onCancel={onCancel}
            onSaveSettings={onSaveSettings}
            formValues={allergiePanelFormValues}
            isBtnShowPanel
            isEditButton={isButtonVisible(hiddenButtons, 'edit', true)}
          >
            <div className="panel-body-inner">
              <div className="form">
                <div className="form-group-wrapper">

                  <div className="row-expand">
                    { isShowElement(valuesNames.CAUSE, hideElements) ?
                      <div className="col-expand-left">
                        <div className="form-group">
                          <label className="control-label">{valuesLabels.CAUSE}</label>
                          <div className="form-control-static">{detail[valuesNames.CAUSE]}</div>
                        </div>
                      </div>
                      : null }
                  { isShowElement(valuesNames.REACTION, hideElements) ?
                      <div className="col-expand-right">
                        <div className="form-group">
                          <label className="control-label">{valuesLabels.REACTION}</label>
                          <div className="form-control-static">{detail[valuesNames.REACTION]}</div>
                        </div>
                      </div>
                      : null }
                  </div>

                  <div className="row-expand">
                    { isShowElement(valuesNames.AUTHOR, hideElements) ?
                        <div className="col-expand-left">
                          <div className="form-group">
                            <label className="control-label">{valuesLabels.AUTHOR}</label>
                            <div className="form-control-static">{detail[valuesNames.AUTHOR]}</div>
                          </div>
                        </div>
                        : null }
                    { isShowElement(valuesNames.CAUSECODE, hideElements) ?
                        <div className="col-expand-right">
                          <div className="form-group">
                            <label className="control-label">{valuesLabels.CAUSECODE}</label>
                            <div className="form-control-static">{detail[valuesNames.CAUSECODE]}</div>
                          </div>
                        </div>
                        : null }
                  </div>

                  <div className="row-expand">
                    { isShowElement(valuesNames.TERMINOLOGY, hideElements) ?
                      <div className="col-expand-left">
                        <div className="form-group">
                          <label className="control-label">{valuesLabels.TERMINOLOGY}</label>
                          <div className="form-control-static">{detail[valuesNames.TERMINOLOGY]}</div>
                        </div>
                      </div>
                    : null }
                  </div>

                  <div className="row-expand">
                    { isShowElement(valuesNames.DATE, hideElements) ?
                        <div className="col-expand-left">
                          <div className="form-group">
                            <label className="control-label">{valuesLabels.DATE_CREATED}</label>
                            <div className="form-control-static">{dateCreated}</div>
                          </div>
                        </div>
                        : null }
                  </div>

                  {detail[valuesNames.ISIMPORT] ?
                    <div className="row-expand">
                      <div className="col-expand-left">
                        <div className="form-group">
                          <label className="control-label">{valuesLabels.ORIGINAL_SOURCE}</label>
                          <div className="form-control-static">{detail[valuesNames.ORIGINAL_SOURCE]}</div>
                        </div>
                      </div>
                      <div className="col-expand-right">
                        <div className="form-group">
                          <label className="control-label">{valuesLabels.ISIMPORT}</label>
                          <div className="form-control-static"></div>
                          <div className="input-holder">
                            <Switch
                              id="isImport"
                              name="isImport"
                              type="checkbox"
                              value
                              chacked
                              disabled
                            />
                          </div>
                        </div>
                      </div>
                    </div> : null
                  }
                </div>
              </div>
            </div>
          </PluginDetailPanel> : null}
          {(expandedPanel === ALLERGIE_PANEL || expandedPanel === 'all') && editedPanel[ALLERGIE_PANEL] ? <PluginDetailPanel
            onExpand={onExpand}
            name={ALLERGIE_PANEL}
            title={detailsTitle}
            onShow={onShow}
            isOpen={openedPanel === ALLERGIE_PANEL}
            currentPanel={currentPanel}
            onEdit={onEdit}
            editedPanel={editedPanel}
            onCancel={onCancel}
            onSaveSettings={onSaveSettings}
            formValues={allergiePanelFormValues}
            isBtnShowPanel
          >
            <AllergyDetailMainForm
              detail={detail}
              isSubmit={isSubmit}
            />
          </PluginDetailPanel> : null }

          {(expandedPanel === SYSTEM_INFO_PANEL || expandedPanel === 'all') && !editedPanel[SYSTEM_INFO_PANEL] && isPanelVisible(hiddenPanels, 'systemInformation') ?
            <PluginDetailPanel
              onExpand={onExpand}
              name={SYSTEM_INFO_PANEL}
              title="System Information"
              isOpen={openedPanel === SYSTEM_INFO_PANEL}
              onShow={onShow}
              currentPanel={currentPanel}
              onEdit={onEdit}
              editedPanel={editedPanel}
              onCancel={onCancel}
              onSaveSettings={onSaveSettings}
              formValues={allergiePanelFormValues}
              isBtnShowPanel
              isShowControlPanel={false}
            >
              <div className="panel-body-inner">
                <div className="form">
                  <div className="form-group-wrapper">
                    <div className="row-expand">
                      { isShowElement(valuesNames.SOURCE, hideElements) ?
                        <div className="col-expand-left">
                          <div className="form-group">
                            <label className="control-label">{valuesLabels.SOURCE}</label>
                            <div className="form-control-static">{detail[valuesNames.SOURCE]}</div>
                          </div>
                        </div>
                        : null }
                    </div>
                  </div>
                </div>
              </div>
            </PluginDetailPanel> : null}
        </div>
      </div>
    )
  }
}
