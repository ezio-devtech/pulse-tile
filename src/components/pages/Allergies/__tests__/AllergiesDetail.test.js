import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import { get } from 'lodash';

import AllergiesDetail from '../AllergiesDetail/AllergiesDetail';
import { valuesNames, valuesLabels } from '../forms.config';
import { getDDMMMYYYY } from '../../../../utils/time-helpers.utils';
import { isShowElement } from '../../../../utils/themeSettings-helper';
import { themeConfigs } from '../../../../themes.config';

Enzyme.configure({ adapter: new Adapter() });

const hideElements = get(themeConfigs, 'detailsToHide.allergies', []);

const propsForAllergiePanel = {
  detail: {
    [valuesNames.CAUSE]: '1111',
    [valuesNames.CAUSECODE]: '1111',
    [valuesNames.TERMINOLOGY]: '1239085',
    [valuesNames.TERMINOLOGYCODE]: '1111',
    [valuesNames.REACTION]: '1111',
    [valuesNames.AUTHOR]: 'Dr Tony Shannon',
    [valuesNames.DATE_CREATED]: 1507020019000,
    [valuesNames.SOURCE]: 'ethercis',
    [valuesNames.SOURCE_ID]: '7ae34463-0770-4a36-bbde-4869bb6a0f05',
    [valuesNames.ORIGINAL_COMPOSITION]: '',
    [valuesNames.ISIMPORT]: true,
    [valuesNames.ORIGINAL_SOURCE]: '',
  },
};

const ALLERGIE_PANEL = 'allergiePanel';
const META_PANEL = 'metaPanel';
const CONVERT_DATE = getDDMMMYYYY(propsForAllergiePanel.detail[valuesNames.DATE_CREATED]);
const SYSTEM_INFO_PANEL = 'systemInformationPanel';

describe('Component <AllergiesDetail />', () => {
  it('should renders with props correctly', () => {
    const component = shallow(<AllergiesDetail />);

    // Testing component when detail filled object, expandedPanel is all, and panel not edited
    component.setProps({ detail: propsForAllergiePanel.detail, expandedPanel: 'all', editedPanel: { [ALLERGIE_PANEL]: false } });
    expect(component.props().className).toEqual('section-detail');
    expect(component.find('PluginDetailPanel')).toHaveLength(2);

    // Testing allergiePanel
    expect(component.find('PluginDetailPanel').at(0).props().name).toEqual(ALLERGIE_PANEL);
    expect(component.find('PluginDetailPanel').at(0).props().title).toEqual('Allergy');
    expect(component.find('PluginDetailPanel').at(0).props().isOpen).toEqual(false);
    expect(component.find('PluginDetailPanel').at(0).props().isBtnShowPanel).toEqual(true);
    expect(component.find('PluginDetailPanel').at(0).props().isShowControlPanel).toEqual(true);

    let countElement = 0;
    if (isShowElement(valuesNames.CAUSE, hideElements)) {
      expect(component.find('.control-label').at(countElement).text()).toEqual(valuesLabels.CAUSE);
      expect(component.find('.form-control-static').at(countElement).text()).toEqual(propsForAllergiePanel.detail.cause);
      countElement++;
    }
    if (isShowElement(valuesNames.REACTION, hideElements)) {
      expect(component.find('.control-label').at(countElement).text()).toEqual(valuesLabels.REACTION);
      expect(component.find('.form-control-static').at(countElement).text()).toEqual(propsForAllergiePanel.detail.reaction);
      countElement++;
    }
    if (isShowElement(valuesNames.AUTHOR, hideElements)) {
      expect(component.find('.control-label').at(countElement).text()).toEqual(valuesLabels.AUTHOR);
      expect(component.find('.form-control-static').at(countElement).text()).toEqual(propsForAllergiePanel.detail.author);
      countElement++;
    }
    if (isShowElement(valuesNames.CAUSECODE, hideElements)) {
      expect(component.find('.control-label').at(countElement).text()).toEqual(valuesLabels.CAUSECODE);
      expect(component.find('.form-control-static').at(countElement).text()).toEqual(propsForAllergiePanel.detail.causeCode);
      countElement++;
    }
    if (isShowElement(valuesNames.TERMINOLOGY, hideElements)) {
      expect(component.find('.control-label').at(countElement).text()).toEqual(valuesLabels.TERMINOLOGY);
      expect(component.find('.form-control-static').at(countElement).text()).toEqual(propsForAllergiePanel.detail.causeTerminology);
      countElement++;
    }
    if (isShowElement(valuesNames.DATE_CREATED, hideElements)) {
       expect(component.find('.control-label').at(countElement).text()).toEqual(valuesLabels.DATE_CREATED);
       expect(component.find('.form-control-static').at(countElement).text()).toEqual(CONVERT_DATE);
       countElement++;
    }
    if (isShowElement(valuesNames.ORIGINAL_SOURCE, hideElements)) {
      expect(component.find('.control-label').at(countElement).text()).toEqual(valuesLabels.ORIGINAL_SOURCE);
      expect(component.find('.form-control-static').at(countElement).text()).toEqual(propsForAllergiePanel.detail.originalSource);
      countElement++;
    }
    if (isShowElement(valuesNames.ISIMPORT, hideElements)) {
      expect(component.find('.control-label').at(countElement).text()).toEqual(valuesLabels.ISIMPORT);
      countElement++;
    }

    // Testing metaPanel
    expect(component.find('PluginDetailPanel').at(1).props().name).toEqual(SYSTEM_INFO_PANEL);
    expect(component.find('PluginDetailPanel').at(1).props().title).toEqual('System Information');
    expect(component.find('PluginDetailPanel').at(1).props().isOpen).toEqual(false);
    expect(component.find('PluginDetailPanel').at(1).props().isBtnShowPanel).toEqual(true);
    expect(component.find('PluginDetailPanel').at(1).props().isShowControlPanel).toEqual(false);

    if (isShowElement(valuesNames.SOURCE, hideElements)) {
      expect(component.find('.control-label').at(countElement).text()).toEqual(valuesLabels.SOURCE);
      expect(component.find('.form-control-static').at(countElement).text()).toEqual(propsForAllergiePanel.detail.source);
    }

    expect(component).toMatchSnapshot();

    component.setProps({ detail: { [valuesNames.ISIMPORT]: true } });

    expect(component.find('Switch')).toHaveLength(1);
  });

  it('should renders correctly with different state of props', () => {
    const component = shallow(
      <AllergiesDetail />);
    // Testing component when detail empty object, expandedPanel is allergiePanel
    component.setProps({ detail: { [valuesNames.DATE_CREATED]: 1507020019000 }, expandedPanel: ALLERGIE_PANEL, editedPanel: { [ALLERGIE_PANEL]: false } });
    expect(component.find('PluginDetailPanel')).toHaveLength(1);
    expect(component.find('PluginDetailPanel').props().name).toEqual(ALLERGIE_PANEL);
    expect(component).toMatchSnapshot();

    // Testing component when detail empty object, expandedPanel is metaPanel
    component.setProps({ detail: {}, expandedPanel: SYSTEM_INFO_PANEL });
    expect(component.find('PluginDetailPanel')).toHaveLength(1);
    expect(component.find('PluginDetailPanel').props().name).toEqual(SYSTEM_INFO_PANEL);
    expect(component).toMatchSnapshot();

    // Testing component when detail filled object, expandedPanel is all, and panel is edited
    component.setProps({ detail: propsForAllergiePanel.detail, expandedPanel: ALLERGIE_PANEL, editedPanel: { [ALLERGIE_PANEL]: true } });
    expect(component.find('ReduxForm')).toHaveLength(1);
    expect(component.find('PluginDetailPanel').props().name).toEqual(ALLERGIE_PANEL);
    expect(component).toMatchSnapshot();
  });
});

