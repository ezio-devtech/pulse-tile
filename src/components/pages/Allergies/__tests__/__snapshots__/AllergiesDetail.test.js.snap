// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <AllergiesDetail /> should renders correctly with different state of props 1`] = `
<div
  className="section-detail"
>
  <div
    className="panel-group accordion"
  >
    <PluginDetailPanel
      editedPanel={
        Object {
          "allergiePanel": false,
        }
      }
      isBtnShowPanel={true}
      isEditButton={true}
      isOpen={false}
      isShowControlPanel={true}
      name="allergiePanel"
      title="Allergy"
    >
      <div
        className="panel-body-inner"
      >
        <div
          className="form"
        >
          <div
            className="form-group-wrapper"
          >
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Cause
                  </label>
                  <div
                    className="form-control-static"
                  />
                </div>
              </div>
              <div
                className="col-expand-right"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Reaction / Description
                  </label>
                  <div
                    className="form-control-static"
                  />
                </div>
              </div>
            </div>
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Author
                  </label>
                  <div
                    className="form-control-static"
                  />
                </div>
              </div>
              <div
                className="col-expand-right"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Cause Code
                  </label>
                  <div
                    className="form-control-static"
                  />
                </div>
              </div>
            </div>
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Terminology
                  </label>
                  <div
                    className="form-control-static"
                  />
                </div>
              </div>
            </div>
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Date
                  </label>
                  <div
                    className="form-control-static"
                  >
                    03-Oct-2017
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </PluginDetailPanel>
  </div>
</div>
`;

exports[`Component <AllergiesDetail /> should renders correctly with different state of props 2`] = `
<div
  className="section-detail"
>
  <div
    className="panel-group accordion"
  >
    <PluginDetailPanel
      editedPanel={
        Object {
          "allergiePanel": false,
        }
      }
      isBtnShowPanel={true}
      isEditButton={true}
      isOpen={false}
      isShowControlPanel={false}
      name="systemInformationPanel"
      title="System Information"
    >
      <div
        className="panel-body-inner"
      >
        <div
          className="form"
        >
          <div
            className="form-group-wrapper"
          >
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Source
                  </label>
                  <div
                    className="form-control-static"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </PluginDetailPanel>
  </div>
</div>
`;

exports[`Component <AllergiesDetail /> should renders correctly with different state of props 3`] = `
<div
  className="section-detail"
>
  <div
    className="panel-group accordion"
  >
    <PluginDetailPanel
      editedPanel={
        Object {
          "allergiePanel": true,
        }
      }
      isBtnShowPanel={true}
      isEditButton={true}
      isOpen={false}
      isShowControlPanel={true}
      name="allergiePanel"
      title="Allergy"
    >
      <ReduxForm
        detail={
          Object {
            "author": "Dr Tony Shannon",
            "cause": "1111",
            "causeCode": "1111",
            "causeTerminology": "1239085",
            "dateCreated": 1507020019000,
            "isImport": true,
            "originalComposition": "",
            "originalSource": "",
            "reaction": "1111",
            "source": "ethercis",
            "sourceId": "7ae34463-0770-4a36-bbde-4869bb6a0f05",
            "terminologyCode": "1111",
          }
        }
      />
    </PluginDetailPanel>
  </div>
</div>
`;

exports[`Component <AllergiesDetail /> should renders with props correctly 1`] = `
<div
  className="section-detail"
>
  <div
    className="panel-group accordion"
  >
    <PluginDetailPanel
      editedPanel={
        Object {
          "allergiePanel": false,
        }
      }
      isBtnShowPanel={true}
      isEditButton={true}
      isOpen={false}
      isShowControlPanel={true}
      name="allergiePanel"
      title="Allergy"
    >
      <div
        className="panel-body-inner"
      >
        <div
          className="form"
        >
          <div
            className="form-group-wrapper"
          >
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Cause
                  </label>
                  <div
                    className="form-control-static"
                  >
                    1111
                  </div>
                </div>
              </div>
              <div
                className="col-expand-right"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Reaction / Description
                  </label>
                  <div
                    className="form-control-static"
                  >
                    1111
                  </div>
                </div>
              </div>
            </div>
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Author
                  </label>
                  <div
                    className="form-control-static"
                  >
                    Dr Tony Shannon
                  </div>
                </div>
              </div>
              <div
                className="col-expand-right"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Cause Code
                  </label>
                  <div
                    className="form-control-static"
                  >
                    1111
                  </div>
                </div>
              </div>
            </div>
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Terminology
                  </label>
                  <div
                    className="form-control-static"
                  >
                    1239085
                  </div>
                </div>
              </div>
            </div>
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Date
                  </label>
                  <div
                    className="form-control-static"
                  >
                    03-Oct-2017
                  </div>
                </div>
              </div>
            </div>
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Original Source
                  </label>
                  <div
                    className="form-control-static"
                  />
                </div>
              </div>
              <div
                className="col-expand-right"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Is Imported
                  </label>
                  <div
                    className="form-control-static"
                  />
                  <div
                    className="input-holder"
                  >
                    <Switch
                      chacked={true}
                      disabled={true}
                      id="isImport"
                      name="isImport"
                      type="checkbox"
                      value={true}
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </PluginDetailPanel>
    <PluginDetailPanel
      editedPanel={
        Object {
          "allergiePanel": false,
        }
      }
      isBtnShowPanel={true}
      isEditButton={true}
      isOpen={false}
      isShowControlPanel={false}
      name="systemInformationPanel"
      title="System Information"
    >
      <div
        className="panel-body-inner"
      >
        <div
          className="form"
        >
          <div
            className="form-group-wrapper"
          >
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Source
                  </label>
                  <div
                    className="form-control-static"
                  >
                    ethercis
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </PluginDetailPanel>
  </div>
</div>
`;
